pkg/metrics/metrics.go

```diff
@@ -0,0 +1,110 @@
+package metrics
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/rancher/apiserver/pkg/apierror"
+)
+
+type MetricLogger struct {
+	Resource string
+	Method   string
+	ID       string
+	Path     string
+}
+
+var prometheusMetrics = false
+
+const (
+	resourceLabel = "resource"
+	methodLabel   = "method"
+	codeLabel     = "code"
+	idLabel       = "id"
```

Comment:
Are we sure we want an id label? If I'm understanding correctly, the value of this label is the resource's `ns/name`

The `/metrics` endpoint will have a row for every unique combination of label values. If we're labeling by id, the cardinality of the set of all possible label values explodes. Requests for different resources will never fall into the same label buckets. And we're doing this for multiple metrics here.

Reply:
Spoke offline, removed id and path label.